code/datums/components/crafting/crafting.dm

```diff
@@ -458,28 +458,41 @@
 
 	return data
 
+/datum/component/personal_crafting/proc/make_action(datum/crafting_recipe/recipe, mob/user)
+	var/atom/movable/result = construct_item(user, recipe)
+	if(!istext(result)) //We made an item and didn't get a fail message
+		if(ismob(user) && isitem(result)) //In case the user is actually possessing a non mob like a machine
+			user.put_in_hands(result)
+		else
+			if(!istype(result, /obj/effect/spawner))
+				result.forceMove(user.drop_location())
+		to_chat(user, span_notice("[recipe.name] crafted."))
+		user.investigate_log("crafted [recipe]", INVESTIGATE_CRAFTING)
+		recipe.on_craft_completion(user, result)
+		return TRUE
+	else
+		to_chat(user, span_warning("Construction failed[result]"))
+		return FALSE
+
+
 /datum/component/personal_crafting/ui_act(action, params)
 	. = ..()
 	if(.)
 		return
 	switch(action)
-		if("make")
+		if("make", "make_mass")
 			var/mob/user = usr
 			var/datum/crafting_recipe/crafting_recipe = locate(params["recipe"]) in (mode ? GLOB.cooking_recipes : GLOB.crafting_recipes)
 			busy = TRUE
 			ui_interact(user)
-			var/atom/movable/result = construct_item(user, crafting_recipe)
-			if(!istext(result)) //We made an item and didn't get a fail message
-				if(ismob(user) && isitem(result)) //In case the user is actually possessing a non mob like a machine
-					user.put_in_hands(result)
-				else
-					if(!istype(result, /obj/effect/spawner))
-						result.forceMove(user.drop_location())
-				to_chat(user, span_notice("[crafting_recipe.name] crafted."))
-				user.investigate_log("crafted [crafting_recipe]", INVESTIGATE_CRAFTING)
-				crafting_recipe.on_craft_completion(user, result)
+			if(action == "make_mass")
+				var/crafted_items = 0
+				while(make_action(crafting_recipe, user))
+					crafted_items++
+				if(crafted_items)
+					to_chat(user, span_notice("You made [crafted_items] item[crafted_items-1 > 1 ? "s" : ""]"))
```

Comment:
```suggestion
					to_chat(user, span_notice("You made [crafted_items] item\s."))
```
This *should* work. Be sure to test it, though.

Reply:
It's working, whoo!